Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go

```diff
@@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
 		}
 	}
 
-	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
+	resourceOrKind := requestInfo.Kind
+	if len(resourceOrKind) == 0 {
+		resourceOrKind = requestInfo.Resource
```

Comment:
So...here's my opinion: in order to not get disabling the features and this larger info handling too coupled I could remove my changes here and just leave the "don't swallow error messages" upstream change.  We'd be left with the ambiguous messages we already have.  WDYT?